truffleruby/src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -839,6 +847,7 @@ private static Frame findCallingMethodFrame() {
                 if (method == null) {
                     return null;
                 } else if (method.getName().equals(/* Truffle::Cext. */ "rb_call_super")
+                        || method.getName().equals(/* Truffle::CExt. */ "execute_with_mutex")
                         || method.getName().equals(/* Truffle::Interop. */ "execute")
                         || method.getName().equals(/* Truffle::Cext. */ "rb_call_super_splatted")) {
                     // TODO CS 11-Mar-17 must have a more precise check to skip these methods
```

Comment:
One good way to fix this is keep references to these InternalMethod and then just compare by identity.
CoreLibrary already looks up some method after boot for the same purpose.